pulsar-client-cpp/include/pulsar/Consumer.h

```diff
@@ -21,13 +21,45 @@
 #include <boost/shared_ptr.hpp>
 #include <pulsar/Message.h>
 #include <pulsar/Result.h>
+#include <boost/date_time/posix_time/ptime.hpp>
 
 #pragma GCC visibility push(default)
 
 class PulsarFriend;
 
 namespace pulsar {
 
+typedef std::map<std::string, std::string> StatsMap;
+
+class BrokerSideConsumerStats {
```

Comment:
There is already a ConsumerStats class in Java client which captures the message rate, acks etc as sent by client -  created by Sangming and Sid.